### PR TITLE
Display alert on change SOGIE fields PROF-43

### PR DESCRIPTION
**Pull request by Sushmareddy-OSI**

### CODE REVIEW

File: app/javascript/Clients/clientIdPage/ClientInformation/Sogie.test.js [hunk 1]
```diff
@@ -24,41 +24,56 @@ describe('Sogie', () => {
...
       const myFunction = instance.handleOnChange('sexual_orintation');
-      expect(() => myFunction({ value: 'something' })).not.toThrow();
-      expect(instance.state.childClient.sexual_orintation).toEqual('something');
+      expect(() => myFunction({ value: 'Something' })).not.toThrow();
+      expect(instance.state.childClient.sexual_orintation).toEqual('Something');
+      expect(instance.state.openAlert).toEqual(false);
+    });
+  });
+
+  describe('#alertPeoceed', () => {
```
Comment (RamuKammagani-osi) [inline app/javascript/Clients/clientIdPage/ClientInformation/Sogie.test.js]: Misspelled Function name

File: app/javascript/Clients/clientIdPage/ClientInformation/Sogie.test.js [hunk 2]
```diff
@@ -24,41 +24,56 @@ describe('Sogie', () => {
...
+          sogie.find('[label="Gender Identity Description(required)"]').exists()
         ).toEqual(true);
       });
     });
   });
 
   describe('#handleOnChange()', () => {
+    it('should handle the event that sets state if value is Not Listed', () => {
+      const instance = sogie.instance();
+      const myFunction = instance.handleOnChange('sexual_orintation');
```
Comment (edgibbs) [inline app/javascript/Clients/clientIdPage/ClientInformation/Sogie.test.js]: I think it should be `sexual_orientation`

File: app/javascript/Clients/clientIdPage/ClientInformation/Sogie.test.js [hunk 3]
```diff
@@ -24,41 +24,56 @@ describe('Sogie', () => {
...
+      expect(() => myFunction({ value: 'Not Listed' })).not.toThrow();
+      expect(instance.state.childClient.sexual_orintation).toEqual(
```
Comment (edgibbs) [inline app/javascript/Clients/clientIdPage/ClientInformation/Sogie.test.js]: same spelling `sexual_orientation`

File: app/javascript/Clients/clientIdPage/ClientInformation/Sogie.test.js [hunk 4]
```diff
@@ -24,41 +24,56 @@ describe('Sogie', () => {
...
+      expect(instance.state.openAlert).toEqual(true);
+    });
+
     it('should handle the event that sets state', () => {
       const instance = sogie.instance();
       const myFunction = instance.handleOnChange('sexual_orintation');
-      expect(() => myFunction({ value: 'something' })).not.toThrow();
-      expect(instance.state.childClient.sexual_orintation).toEqual('something');
+      expect(() => myFunction({ value: 'Something' })).not.toThrow();
+      expect(instance.state.childClient.sexual_orintation).toEqual('Something');
```
Comment (edgibbs) [inline app/javascript/Clients/clientIdPage/ClientInformation/Sogie.test.js]: `sexual_orientation`

**Pull request merged**